packages/rest/src/rest-server.ts

```diff
@@ -78,7 +78,7 @@ export class RestServer extends Context implements Server {
 
   constructor(
     @inject(CoreBindings.APPLICATION_INSTANCE) app: Application,
-    @inject(RestBindings.CONFIG) options?: RestServerConfig,
+    options?: RestServerConfig,
```

Comment:
Mixed usage of injected and non-injected parameters is suspicious to me. Can you explain?

Reply:
We can't inject the config using a shared key, because then every server of the same type will get the same config (since there's only one app context).

Reply:
1. Does it hint that each server instance should have its own context as a child of app?
2. Is RestServer not intended for DI any more? If so, we should get rid of `@inject(CoreBindings.APPLICATION_INSTANCE)` too.

Reply:
>Does it hint that each server instance should have its own context as a child of app?

Each server instance already *does* have its own context, in which the Application is its parent.

>Is RestServer not intended for DI any more? If so, we should get rid of @inject(CoreBindings.APPLICATION_INSTANCE) too.

The RestServer is still taking advantage of DI. Each server instance will have its values bound to keys using that particular instance's `.bind` function.

The reason why its constructor doesn't take injected values for the config is that would have to get its injected value from the app-level binding. That's totally fine for injecting *itself* into the instance but if the configuration for the servers is done in the same way, then you're limited to one instance (since they're all going to get the same config).

Reply:
See my older https://github.com/strongloop/loopback-next/pull/605#issuecomment-333815352 for an example solution showing how to keep all server constructor arguments injectable and still support multiple different server configs.